### ISSUE TITLE
Use Dittofeed for in-app messages

### ISSUE THREAD

**Issue by pcr0wley** [crowleysystems/dittofeed#6]
It would be nice to use Dittofeed to power in-app messages, similar to Customer.io.

**Customer.io**
https://customer.io/docs/journeys/send-in-app-message

It looks like it is on their future roadmap, however I am not sure of the timeframe.
https://docs.dittofeed.com/integrations/channels

@SergioPSV Can you check with Max and see what their timeframe on in-app messages is?

_This is a complicated problem and I am not sure if it would be available in the near future._

**Comment by SergioPSV** [comment]
@pcr0wley I received a response from Max

> We plan to support them via simple webhooks to e.g. firebase cloud messaging in the near future. This which should cover the majority of use cases, but will be lacking more advanced features, and tracking (e.g. deeplinking supporting). This should be available sometime within the next two weeks.
For more advanced use cases, we'll be releasing dedicated native mobile sdk's which support push notifications. These represent a larger lift and are several months down the line.

There are no plans for in-app messages yet, but with the help of the SDK it will be possible (this will not be soon, an SDK with push notification support will be available first)

I also see that we use push notification in customer.io only for iOS using the Apple Push Notification service. Dittofeed will be adding push notification capability within a few weeks https://firebase.google.com/docs/cloud-messaging, but we will need to configure the application for this (for reference, customer.io also has support for Firebase Cloud Messaging, but we don't use it)

**Comment by pcr0wley** [comment]
@SergioPSV Sounds good.

It looks like Firebase supports web push notifications through standard browser functionality, but not in-app messages.

Basically this means that all notifications to the website would first require user permission, and would not be customizable.

Can you ask Max what they recommend if we want to do popup notifications on our website, and not just push notifications through the browser?

**Comment by SergioPSV** [comment]
@pcr0wley Yes, push notifications will require permission in the browser, but this is more necessary for notifications in ios and android applications. Firebase also has https://firebase.google.com/docs/in-app-messaging, but this too is mostly used in apps.

Perhaps https://tally.so/help/popup-forms will be suitable for our purposes. Using this form, we will be able to flexibly configure any notifications on the site and automatically hide them if they do not interest the user, and they can also be edited without using code.

**Comment by pcr0wley** [comment]
Regarding popup forms, if we used an external service we would need to find something that accepted webhooks from Dittofeed.

**Comment by SergioPSV** [comment]
We can do popup notifications on our website when the SDK is ready, but Max can't say exactly when it will be ready. I previously indicated that we need this function within a month, but as I understand it, it’s better for us to look for other options. And in a few months we will be able to use the SDK
